AmplifyPlugins/Analytics/AWSPinpointAnalyticsPlugin/Dependency/Pinpoint/Endpoint/Device.swift

```diff
@@ -0,0 +1,37 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+protocol Device {
+    typealias Platform = (name: String, version: String)
+    
+    var model: String { get }
+    var appVersion: String? { get }
+    var platform: Platform { get }
+}
+
+#if canImport(UIKit)
+import UIKit
+
+extension UIDevice: Device {
+    var appVersion: String? {
+        return Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String
```

Comment:
nit: there's an existing extension for `Bundle.appVersion` defined in PinpointEvent+PinpointClientTypes.swift

Reply:
Oh sweet! I missed it